R/assertions.R

```diff
@@ -47,21 +58,23 @@ assert_data_frame <- function(arg,
     return(invisible(arg))
   }
 
-  if (!is.data.frame(arg)) {
-    err_msg <- sprintf(
-      "`%s` must be a data frame but is %s",
-      arg_name(substitute(arg)),
-      what_is_it(arg)
-    )
-    abort(err_msg)
-  }
+  assert_s3_class(
+    arg,
+    cls = "data.frame",
+    optional = optional,
+    arg_name = arg_name,
+    message = message,
+    class = class,
+    call = call
+  )
 
   if (check_is_grouped && dplyr::is_grouped_df(arg)) {
-    err_msg <- sprintf(
-      "`%s` is a grouped data frame, please `ungroup()` it first",
-      arg_name(substitute(arg))
+    cli_abort(
+      message = message %||%
+        "Argument {.arg {arg_name}} must not be grouped, please `ungroup()` it first",
```

Comment:
```suggestion
        "Argument {.arg {arg_name}} must not be a grouped data frame, please `ungroup()` it.",
```

Reply:
what if it is a tibble? :) maybe _must not be a grouped dataset..._

Reply:
Updated